### PR TITLE
Bump next

### PR DESCRIPTION
**Pull request by Joozty**
@viktor-yakubiv As I already told you the biggest change was renaming `static` folder to public. This affected the output of `yarn export` command. I changed `public` to dist. I assume we will need to adjust our deployment too.. 

Also, [app container is deprected](https://github.com/zeit/next.js/blob/canary/errors/app-container-deprecated.md) therefore I removed it from _app.js

### CODE REVIEW

File: components/layout/document.jsx [hunk 1]
```diff
@@ -1,27 +1,5 @@
...
-    _documentProps: PropTypes.any,
-  }
-
-  render() {
-    const { html } = this.context._documentProps
-    const { className = '', ...restProps } = this.props
-    return (
-      <div
-        id="__next"
-        className={`page ${className}`}
```
Comment (viktor-yakubiv) [inline components/layout/document.jsx]: I was passing additional class name what was necessary if I remember. How does it work now? I could not find it in the diffs.

File: babel.config.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module.exports = api => {
```
Comment (viktor-yakubiv) [inline babel.config.js]: What is API?

File: package.json [hunk 3]
```diff
@@ -59,13 +59,13 @@
...
+    "test-dev": "blc http://localhost:3000 -ro --exclude='/browse' --exclude='/search' --exclude='/public' --exclude='/article-update' --exclude='/api-keys' --exclude='/documentation/api' --exclude='/dataset/getdatadump' --exclude='/recommender/register' --exclude='/images/harvested-by-core-orange.png' --exclude='/images/powered-by-core-orange.png'",
     "test-local": "yarn test-dev --exclude-external",
-    "test-staging": "blc https://${CORE_STAGING_AREA}core.ac.uk -ro --exclude='/browse' --exclude='/search' --exclude='/static'",
+    "test-staging": "blc https://${CORE_STAGING_AREA}core.ac.uk -ro --exclude='/browse' --exclude='/search' --exclude='/public'",
     "test": "yarn test-dev",
     "dev": "next",
     "build": "next build",
-    "export": "NEXT_TARGET=server next build && NEXT_TARGET=server next export -o public",
+    "export": "NEXT_TARGET=server next build && NEXT_TARGET=server next export -o dist",
```
Comment (viktor-yakubiv) [inline package.json]: Let's use default directory instead `dist`. We need to add `out` to `.gitignore` and remove `-o dist` here.

File: components/layout/head.jsx [hunk 4]
```diff
@@ -17,9 +17,9 @@ const Head = ({ title, description, keywords, children, ...restProps }) => (
       />
     )}
 
-    <link rel="shortcut icon" href="/static/favicon/favicon-32px.png" />
-    <link rel="icon" href="/static/favicon/favicon-128px.png" />
-    <link rel="icon" href="/static/favicon/favicon.svg" />
+    <link rel="shortcut icon" href="/public/favicon/favicon-32px.png" />
+    <link rel="icon" href="/public/favicon/favicon-128px.png" />
+    <link rel="icon" href="/public/favicon/favicon.svg" />
```
[the remark this answers is not in the document]
Comment (Joozty) [reply components/layout/head.jsx]: yes indeed, thanks

File: data/services/dataset.yml [hunk 5]
```diff
@@ -48,15 +48,15 @@ additional:
...
+    commercial purposes, including consultancies and commercial analysis, please [contact](~about/#contact) us.
     The dataset provides you with:
 
     * The entire CORE's corpus of both metadata and full texts in a machine processable format.
-    * Mappings of the CORE articles to entities in the Micrsoft Academic Graph (MAG), enabling to access 
-      CORE fulltexts and use additional entities from MAG where available. 
+    * Mappings of the CORE articles to entities in the Micrsoft Academic Graph (MAG), enabling to access
+      CORE fulltexts and use additional entities from MAG where available.
     * Detailed [documentation](~documentation/dataset) on how to download the CORE dataset and how data is organised.
-
```
[the remark this answers is not in the document]
Comment (Joozty) [reply data/services/dataset.yml]: seems eslint did this

File: components/layout/document.jsx (hunk 1 above)
Comment (Joozty) [reply components/layout/document.jsx]: Me neither. I just check whether you pass down an additional class name. I didn't find any usage, therefore, I removed it.

File: components/key-feature/index.jsx [hunk 6]
```diff
@@ -13,17 +13,12 @@ const KeyFeature = ({
...
-      { className: `key-feature ${className} ${contentProps.className}` },
-      restProps
-    )
-  }
+  const contentProps =
+    ContentTag !== Fragment
+      ? {
+          className: 'key-feature-link',
+        }
+      : restProps
```
[the remark this answers is not in the document]
Comment (Joozty) [reply components/key-feature/index.jsx]: The warning was about className is not allowed for Fragment. So this should prevent it. I don't know what can be in restProps so I rather left it there to not break anything. I was thinking that maybe there is a `key`. Not sure if it is even possible.

[the remark this answers is not in the document]
Comment (viktor-yakubiv) [reply components/key-feature/index.jsx]: Okay if it works

File: package.json (hunk 3 above)
Comment (Joozty) [reply package.json]: https://github.com/oacore/next/pull/133/commits/b0dd3feb8bc637a01b94811bde92909fa2d1fad1

File: components/layout/head.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Joozty) [reply components/layout/head.jsx]: https://github.com/oacore/next/pull/133/commits/b0dd3feb8bc637a01b94811bde92909fa2d1fad1

File: components/footer/index.jsx [hunk 7]
```diff
@@ -43,7 +43,7 @@ const Footer = ({ usefulLinks, partners, researchOutputs, className = '' }) => (
           target="_blank"
           rel="noopener noreferrer"
         >
-          <img src="/static/images/logos/jisc.svg" alt="Jisc" />
+          <img src="/images/logos/jisc.svg" alt="Jisc" />
```
[the remark this answers is not in the document]
Comment (Joozty) [reply components/footer/index.jsx]: https://github.com/oacore/next/pull/133/commits/b0dd3feb8bc637a01b94811bde92909fa2d1fad1

File: data/services/dataset.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (viktor-yakubiv) [reply data/services/dataset.yml]: WebStorm does this by default. It's a problem in this file

File: babel.config.js (hunk 2 above)
Comment (Joozty) [reply babel.config.js]: it allows you to adjust some babel settings like cache

https://babeljs.io/docs/en/config-files#config-function-api

File: data/services/dataset.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Joozty) [reply data/services/dataset.yml]: I was wrong. It's not linter but `.editorconfig`. I guess this is that rule https://github.com/oacore/next/blob/content/.editorconfig#L11

**Pull request merged**